_posts/2024-10-03-pixi-basics.md

```diff
@@ -0,0 +1,358 @@
+---
+layout: post
+title: Pixi basics
+date: 2024-10-03 15:00:00
+author: "Lee Katz"
+tags:
+  - development
+  - conda
+  - bioconda
+  - software
+  - pixi
+contributors:
+  - lskatz
+---
+
+Recently I had an issue where Conda was no longer a viable solution
+for my day to day.
+Namely 1) the defaults channel was becoming a for-pay model and 2) the actual site anaconda.org was being blocked at work.
+
+Pixi is an alternative to conda where it is really easy to remove the defaults channel and where it is a slightly different model:
+Each time you "activate" the environment, it is actually an environment meant for that specific directory.
+Therefore, a project can have a designated environment that gets activated with `pixi init` when you or a collaborator enter the directory and wants to get started.
+
+Also, the whole thing is written in Rust which I really really respect for stability and speed.
+
+## Starting off
+
+The only way I can think of how to show this off properly is to get a real example.
+I'll choose a random software package hmmm.... `mashtree`?
+
+### Configuring
+
+First, if your system is like mine, you have a few things to change:
+remove anaconda and account for an older system.
+
+First, start off the config file. It lives in `~/.config/pixi`
```

Comment:
Perhaps worth pointing out that a system or user-wide config is not strictly needed, but might avoid re-typing this every time you create a new env.

Reply:
Changing to

> First, start off the config file in your user space. It lives in `~/.config/pixi`. _Note_: you can have [system-wide configs or local configs](https://pixi.sh/latest/reference/pixi_configuration/) but I will leave that outside the scope of this post.